### PR TITLE
ref(db): Explicitly use replica for platform stats task

### PR DESCRIPTION
**Pull request by beezz**

### CODE REVIEW

**Review by markstory — APPROVED** [review]
Looks good to me. The router database selection will fallback to the primary servers in self-hosted environments.

**Review by wedamija — APPROVED** [review]
If this is something we're going to do a lot then it might make sense to implement `.using_replica` in our base manager, which would be something like

```
def using_replica(self):
    return self.using(router.db_for_read(self.model, replica=True))
```

**Comment by beezz** [comment]
> If this is something we're going to do a lot then it might make sense to implement `.using_replica` in our base manager, which would be something like
> 
> ```
> def using_replica(self):
>     return self.using(router.db_for_read(self.model, replica=True))
> ```

I added `using_replica` to the base queryset class  so it can be used on existing querysets, chained after filter/exclude etc.
If called on manager (Model.objects) it still works cause manager just proxy all methods to queryset class.

Not sure what to do with the typing issue yet 🙃

File: src/sentry/db/models/manager/base.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 _local_cache_enabled = False
 
 
-class BaseManager(Manager, Generic[M]):  # type: ignore
+class BaseManager(DjangoBaseManager.from_queryset(BaseQuerySet), Generic[M]):  # type: ignore
```
Comment (beezz) [inline src/sentry/db/models/manager/base.py:27]: https://github.com/django/django/blob/stable/2.2.x/django/db/models/manager.py#L165

**Comment by markstory** [comment]
> Not sure what to do with the typing issue yet 🙃

I wouldn't worry too much about the typing errors. Our django model types are quite wrong and emit typing errors everywhere.

**Review by wedamija — APPROVED** [review]

**Pull request merged**